pystats_docs.md

```diff
@@ -0,0 +1,81 @@
+# pystats documentation
+
+This describes the meaning of the various sections and fields in the pystats output.
+
+## Execution counts
+
+This is the counts of how many times each Tier 1 instruction is executed.
+
+The "miss ratio" column shows the percentage of times the instruction executed that it deoptimized.
+
+## Pair counts
+
+Shows the Top 100 most frequently occurring pairs of instructions as executed.
+
+## Predecessor / Successor pairs
+
+Shows the top 5 instructions that are executed before and after each instruction.
+
+## Specialization stats
+
+TBD
+
+## Specialization effectiveness
+
+TBD
+
+## Call stats
+
+TBD
+
+## Object stats
+
+TBD
+
+## GC stats
+
+TBD
+
+## Optimization (Tier 2) stats
+
+- Optimization attempts: The number of times a trace is found, and attempt is
+  made to optimize and run it in Tier 2. From the JUMP_BACKWARD instruction when
+  the counter reaches `optimizer_backedge_threshold`.
+- Traces created: The number of traces that were successfully created.
+- Trace stack overflow: A potential trace is abandoned because it would require
```

Comment:
Those traces aren’t really abandoned — if long enough they will still be created and executed. Maybe the thing this counts is how many times trace collection ended for that reason. Same for the next ones, except “too short”, which really does abandon it (no executor is created).